addons/account_payment/tests/test_account_payment.py

```diff
@@ -296,3 +296,48 @@ def test_payment_invoice_same_receivable(self):
 
         self.assertNotEqual(self.partner.property_account_receivable_id, payment.destination_account_id)
         self.assertEqual(payment.destination_account_id, invoice.line_ids[-1].account_id)
+        
+    def test_vendor_payment_name_remains_same_after_repost(self):
+            partner = self.env['res.partner'].create({
+                'name': 'Test Vendor',
+                'supplier_rank': 1,
+            })
+
+            journal = self.env['account.journal'].search([('type', '=', 'bank')], limit=1)
+            self.assertTrue(journal, "You need at least one bank journal for this test.")
```

Comment:
You are in a test case, use
`journal = self.company_data['default_journal_bank']`